linker/Linker.Steps/MarkStep.cs

```diff
@@ -822,10 +834,8 @@ protected void MarkField (FieldReference reference)
 			DoAdditionalFieldProcessing (field);
 
 			var parent = reference.DeclaringType.Resolve ();
-			if (parent.IsBeforeFieldInit && !Annotations.HasPreservedStaticCtor (parent) && _context.IsOptimizationEnabled (CodeOptimizations.BeforeFieldInit)) {
-				if (MarkMethodIf (parent.Methods, IsStaticConstructor))
-					Annotations.SetPreservedStaticCtor (parent);
-			}
+			if (!Annotations.HasPreservedStaticCtor (parent))
```

Comment:
Shouldn't this be `ShouldMarkTypeStaticConstructor` ?

Reply:
All the tests pass 😄 

I wasn't sure.  I actually had `ShouldMarkTypeStaticConstructor` at first.  But then the `StaticFieldInitializer` test fails.